### PR TITLE
Feature/nijo 1013 load contract file

### PR DESCRIPTION
**Pull request by Raibo**

### CODE REVIEW

File: Api/Services/Files/ContractFileService.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        {
+            return await ValidateFile()
+                .Bind(Upload);
+
+
+            Result ValidateFile() =>
+                Result.Success()
+                    .Ensure(() => file != null, "Couldn't get any file")
+                    .Ensure(() => file.Length > 0, "Got an empty file")
+                    .Ensure(() => Path.GetExtension(file?.FileName)?.ToLower() == ".pdf", "The file must have extension '.pdf'");
```
Comment (mingazhev) [inline Api/Services/Files/ContractFileService.cs:34]: this should be extracted to constants

Comment (mingazhev) [inline Api/Services/Files/ContractFileService.cs:34]: string.Equals with stringComparison.OrdinalIgnoreCase will be shorter

File: Api/Services/Files/ContractFileService.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            {
+                await using var stream = file.OpenReadStream();
+                return await _amazonS3ClientService.Add(_bucketName, GetKey(counterpartyId), stream, S3CannedACL.Private);
+            }
+        }
+
+
+        public async Task<Result<Stream>> Get(int counterpartyId)
+        {
+            var (_, isFailure, stream, _) = await _amazonS3ClientService.Get(_bucketName, GetKey(counterpartyId));
```
Comment (mingazhev) [inline Api/Services/Files/ContractFileService.cs:47]: Why not in pipeline format? Will be better if top-level methods has the same style

File: Api/Services/Files/ContractFileService.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+            if (isFailure)
+                return Result.Failure<Stream>("Couldn't get a contract file");
+
+            return Result.Success(stream);
```
Comment (mingazhev) [inline Api/Services/Files/ContractFileService.cs:52]: looks like Result.Success could be omitted

File: Api/Controllers/AdministratorControllers/CounterpartiesController.cs [hunk 4]
```diff
@@ -173,6 +177,43 @@ public async Task<IActionResult> GetCounterpartyPredictions(string query)
...
+        [ProducesResponseType(typeof(ProblemDetails), (int)HttpStatusCode.BadRequest)]
+        [AdministratorPermissions(AdministratorPermissions.CounterpartyManagement)]
+        public async Task<IActionResult> GetContractFile(int counterpartyId)
+        {
+            var (_, isFailure, stream, error) = await _contractFileService.Get(counterpartyId);
+
+            if (isFailure)
+                return BadRequest(ProblemDetailsBuilder.Build(error));
+
+            return File(stream, "application/pdf");
```
Comment (mingazhev) [inline Api/Controllers/AdministratorControllers/CounterpartiesController.cs:212]: Is getting the strict contact type is responsibility of controller?
I think controller should not know about the file format, this logic is bound to a service

File: Api/Services/Files/ContractFileService.cs (hunk 1 above)
Comment (Raibo) [reply Api/Services/Files/ContractFileService.cs:34]: `string.Equals(Path.GetExtension(file?.FileName), PdfFileExtension, StringComparison.OrdinalIgnoreCase)`
vs
`Path.GetExtension(file?.FileName)?.ToLower() == PdfFileExtension`
Not shorter at all

File: Api/Services/Files/ContractFileService.cs (hunk 2 above)
Comment (Raibo) [reply Api/Services/Files/ContractFileService.cs:47]: There is simply nothing to put into local methods, we simply call the other service and swallow possible error, nothing more.

File: Api/Services/Files/ContractFileService.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Files/ContractFileService.cs]: ok, so use tolowerinvariant to discard the culture, this is a good practice

**Review by mingazhev — APPROVED** [review]
with a little comment

**Pull request merged**